Move LoadAddrman from init to addrdb

maflcko opened the pull request:
Init should only concern itself with the initialization order, not the detailed initialization logic of every module.

Discussed in https://github.com/bitcoin/bitcoin/pull/22697#discussion_r691310217

Commit:
refactor: Fix includes (add missing ones, remove unneeded)

laanwj commented on src/addrman.cpp:
```diff
@@ -733,3 +736,22 @@ std::vector<bool> CAddrMan::DecodeAsmap(fs::path path)
+
+void LoadAddrman(std::unique_ptr<CAddrMan>& addrman, const ArgsManager& args)
+{
+    assert(!addrman);
```
Maybe return `std::unique_ptr<CAddrMan>`? (or an optional, if it can fail, but doesn't seems so)

MarcoFalke replied:
Done

vasild commented:
Concept ACK

Commit:
Move LoadAddrman from init to addrdb

Init should only concern itself with the initialization order, not the
detailed initialization logic of every module.

jonatack commented on src/addrman.cpp:
```diff
@@ -733,3 +736,22 @@ std::vector<bool> CAddrMan::DecodeAsmap(fs::path path)
+    addrman = std::make_unique<CAddrMan>(/* deterministic */ false, /* consistency_check_ratio */ check_addrman);
+
+    // Load addresses from peers.dat
+    int64_t nStart = GetTimeMillis();
```
if you retouch
```suggestion
    const int64_t nStart{GetTimeMillis()};
```

MarcoFalke replied:
Done

jonatack reviewed (commented):
Concept ACK

jnewbery commented:
I'm ~0 on this change. I don't think there's any real benefit in moving this initialization code from init.cpp to a free function in addrdb.cpp that gets called by init. That doesn't seem to me to be reducing the complexity of what `AppInitMain()` is doing, rather than just hiding it behind a function call.

MarcoFalke commented:
init.cpp is almost 2000 lines of code. I don't see why init.cpp needs to include all existing headers to learn about the implementation details of each module to initialize them. Just like wallet initialization and block importing was moved out, I think the address database details can be moved out as well.

jnewbery commented:
> I don't see why init.cpp needs to include all existing headers to learn about the implementation details of each module to initialize them.

This PR doesn't reduce the number of headers that init.cpp includes (and I also don't think it's a problem for init to include lots of headers - after all, it's init's job to start everything up so it needs to include the headers for all the submodules).

> Just like wallet initialization and block importing was moved out, I think the address database details can be moved out as well.

The difference with wallet initialization and block importing is that those are internal to the wallet and validation modules. Addrman initialization from peers.dat is something that's being done to a CAddrMan externally, even after this PR.

To be clear, I don't object to this and I'm certainly not NACKing, I just don't think it brings much benefit.

MarcoFalke commented:
As this is required for some other changes, I am just going to bundle with them.
